# Custom tweaks
# Fix locale support utf-8
--- src/main.c	(Revision 720)
+++ src/main.c	(Arbeitskopie)
@@ -107,12 +107,17 @@
   gchar *display_env;
 #endif
 
+
 #ifdef ENABLE_NLS
+  setlocale (LC_ALL, "");
   bindtextdomain (PACKAGE, PACKAGE_LOCALE_DIR);
+  bind_textdomain_codeset (PACKAGE, "utf-8");
   textdomain (PACKAGE);
 #endif
 
+  /* Deprecated in favour of setlocale(3)
   gtk_set_locale ();
+  */
 
 #ifdef DEVEL_CODE
   g_print (_("WARNING: Build includes incomplete development code.\n"));

# Modify Super_L to Alt_L
--- src/sample-display.c	2018-04-09 12:01:53.000000000 +0200
+++ src/sample-display.c	2018-04-09 12:03:06.491415168 +0200
@@ -2440,7 +2440,7 @@
 
   switch (event->keyval) {
   case GDK_Meta_L:
-  case GDK_Super_L:
+  case GDK_Alt_L:
   case GDK_Multi_key:
     if (sample->edit_mode == SWEEP_EDIT_MODE_ALLOC) break;
 
@@ -2659,7 +2659,7 @@
 
   switch (event->keyval) {
   case GDK_Meta_L:
-  case GDK_Super_L:
+  case GDK_Alt_L:
   case GDK_Multi_key:
     s->meta_down = FALSE;
 
# Hide "Show history submenu"
--- src/view.c	2018-04-09 22:05:55.000000000 +0200
+++ src/view.c	2018-04-09 22:08:03.854254436 +0200
@@ -536,10 +536,6 @@
                                                   redo_cb, TRUE,
 												  GDK_r, GDK_CONTROL_MASK, view);
 
-  create_view_menu_item (submenu, _("Show history ..."), "<Sweep-View>/Edit/Show history ...", view,
-                                                  show_undo_dialog_cb, FALSE,
-												  0, 0, view);
-
   menuitem = gtk_menu_item_new(); /* Separator */
   gtk_menu_append(GTK_MENU(submenu), menuitem);
   gtk_widget_show(menuitem);

# Switch from aiff to wav
--- src/sweep_sample.c	2018-04-09 22:27:50.000000000 +0200
+++ src/sweep_sample.c	2018-04-09 22:28:57.999316183 +0200
@@ -116,7 +116,7 @@
 static void
 filename_generate (char *name, size_t len)
 {
-  snprintf (name, len, "%s-%d.aiff", _("Untitled"), ++untitled_count);
+  snprintf (name, len, "%s-%d.wav", _("Untitled"), ++untitled_count);
 }
 
 sw_sample *
