--- a/configure.ac	2014-12-09 20:37:27.000000000 +0100
+++ b/configure.ac	2019-04-17 14:56:33.605262745 +0200
@@ -98,6 +98,13 @@
   AC_CHECK_LIB(gmodule-2.0, g_module_init, HAVE_GMODULE="maybe", sweep_config_ok="no")
 fi
 
+PKG_CHECK_MODULES(GMODULE, gmodule-2.0, HAVE_GMODULE="yes", sweep_config_ok="no")
+AC_SUBST(GMODULE_CFLAGS)
+AC_SUBST(GMODULE_LIBS)
+if test "x$HAVE_GMODULE" != xyes ; then
+  AC_CHECK_LIB(gmodule-2.0, g_module_open, HAVE_GMODULE="maybe", sweep_config_ok="no")
+fi
+
 dnl
 dnl  Detect libsndfile 1.0
 dnl
@@ -506,13 +513,13 @@
 *** misconfigured on your system:
 ])
 
-if test "x$HAVE_GLIB" != xyes ||  test "x$HAVE_GTHREADS" != xyes; then
+if test "x$HAVE_GLIB" != xyes || test "x$HAVE_GTHREADS" != xyes || test "x$HAVE_GMODULE" != xyes; then
   AC_MSG_RESULT([
 *** GLib version 2.2.x or greater, available from  http://www.gtk.org/
 ])
 fi
 
-if test "x$HAVE_GLIB" = xmaybe || test "x$HAVE_GTHREADS" = xmaybe; then
+if test "x$HAVE_GLIB" = xmaybe || test "x$HAVE_GTHREADS" = xmaybe || test "x$HAVE_GMODULE" = xmaybe; then
   AC_MSG_RESULT([
       Development files missing: The libglib-2.0 binary seems
       to be installed, but the pkg-config tool cannot find its build
@@ -573,6 +580,7 @@
 if test "x$HAVE_GLIB" = xmaybe || test "x$HAVE_GTK" = xmaybe ||
    test "x$HAVE_LIBSNDFILE" = xmaybe || test "x$HAVE_TDB" = xmaybe ||
    test "x$HAVE_GTHREADS" = xmaybe || test "x$HAVE_GMODULE" = xmaybe ||
+   test "x$HAVE_ALSA" = xmaybe ||
    test "x$HAVE_OGG" = xmaybe || test "x$HAVE_VORBIS" = xmaybe ; then
   AC_MSG_RESULT([
 *** HOW TO FIX "development files missing": you have probably


--- a/src/Makefile.am.orig	2019-04-17 14:53:29.600253685 +0200
+++ b/src/Makefile.am	2019-04-17 14:57:42.582266141 +0200
@@ -11,6 +11,7 @@
 	@GTK_CFLAGS@ \
 	@GLIB_CFLAGS@ \
 	@GTHREADS_CFLAGS@ \
+	@GMODULE_CFLAGS@ \
 	@ALSA_CFLAGS@ \
 	@PULSEAUDIO_CFLAGS@
