From 513999baf919d255fa12e03daa74b276fb8c4409 Mon Sep 17 00:00:00 2001
From: jadiah <20830633+jadiah@users.noreply.github.com>
Date: Fri, 12 Apr 2019 12:20:23 +0300
Subject: [PATCH] Added "Best Fit" option to "PopupMenu"

---
 src/vnr-window.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vnr-window.c b/src/vnr-window.c
index 435b45d..3ebda28 100755
--- a/src/vnr-window.c
+++ b/src/vnr-window.c
@@ -187,6 +187,7 @@
     "<menuitem action=\"ViewZoomIn\"/>"
     "<menuitem action=\"ViewZoomOut\"/>"
     "<menuitem action=\"ViewZoomNormal\"/>"
+	"<menuitem action=\"ViewZoomFit\"/>"
     "<separator/>"
     "<menuitem name=\"MenuBar\" action=\"ViewMenuBar\"/>"
     "<menuitem name=\"Toolbar\" action=\"ViewToolbar\"/>"

@@ -1725,7 +1726,7 @@
     { "ViewZoomNormal", GTK_STOCK_ZOOM_100, N_("_Normal Size"), "<control>0",
       N_("Show the image at its normal size"),
       G_CALLBACK (vnr_window_cmd_normal_size) },
-    { "ViewZoomFit", GTK_STOCK_ZOOM_FIT, N_("Best _Fit"), NULL,
+    { "ViewZoomFit", GTK_STOCK_ZOOM_FIT, N_("Best _Fit"), "<control>1",
       N_("Fit the image to the window"),
       G_CALLBACK (vnr_window_cmd_fit) },
     { "ControlEqual", GTK_STOCK_ZOOM_IN, N_("_Zoom In"), "<control>equal",
