--- a/src/main2.c	2018-05-13 11:40:40.000000000 +0200
+++ b/src/main.c	2021-09-29 14:30:17.072912794 +0200
@@ -112,10 +112,7 @@
         }
         else if(file_list == NULL)
         {
-            deny_slideshow(VNR_WINDOW(window));
-            vnr_message_area_show(VNR_MESSAGE_AREA (VNR_WINDOW(window)->msg_area),
-                                  TRUE, _("The given locations contain no images."),
-                                  TRUE);
+            gtk_main_quit();
         }
         else
         {

--- a/src/vnr-window2.c	2018-05-13 11:40:40.000000000 +0200
+++ b/src/vnr-window.c	2021-09-29 14:29:36.625913993 +0200
@@ -1656,18 +1656,7 @@
 
             if(next == NULL)
             {
-                vnr_window_close(window);
-                gtk_action_group_set_sensitive(window->actions_collection, FALSE);
-                deny_slideshow(window);
-                vnr_window_set_list(window, NULL, FALSE);
-                vnr_message_area_show(VNR_MESSAGE_AREA (window->msg_area), TRUE,
-                                      _("The given locations contain no images."),
-                                      TRUE);
-                restart_slideshow = FALSE;
-
-
-                if(gtk_widget_get_visible(window->props_dlg))
-                    vnr_properties_dialog_clear(VNR_PROPERTIES_DIALOG(window->props_dlg));
+                gtk_main_quit();
             }
             else
             {
