From 7bedde67cff7e44557ad361bc08f0b97bdecaa8d Mon Sep 17 00:00:00 2001
From: Nick Cao <nickcaothecoder@gmail.com>
Date: Mon, 25 Nov 2019 20:48:44 +0800
Subject: [PATCH] Remove the ipv4 address shortcut and adds an any address
 shortcut.

---
 src/AsyncNameResolverMan.cc | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/AsyncNameResolverMan.cc b/src/AsyncNameResolverMan.cc
index be0735614..0d48e030d 100644
--- a/src/AsyncNameResolverMan.cc
+++ b/src/AsyncNameResolverMan.cc
@@ -151,14 +151,10 @@ int AsyncNameResolverMan::getStatus() const
 {
   size_t success = 0;
   size_t error = 0;
-  bool ipv4Success = false;
   for (size_t i = 0; i < numResolver_; ++i) {
     switch (asyncNameResolver_[i]->getStatus()) {
     case AsyncNameResolver::STATUS_SUCCESS:
       ++success;
-      if (asyncNameResolver_[i]->getFamily() == AF_INET) {
-        ipv4Success = true;
-      }
       break;
     case AsyncNameResolver::STATUS_ERROR:
       ++error;
@@ -167,12 +163,7 @@ int AsyncNameResolverMan::getStatus() const
       break;
     }
   }
-  // If we got a IPv4 lookup response, we don't wait for a IPv6 lookup
-  // response. This is because DNS servers may drop AAAA queries and we
-  // have to wait for a long time before timeout. We don't do the
-  // inverse, because, based on today's deployment of DNS servers,
-  // almost all of them can respond to A queries just fine.
-  if ((success && ipv4Success) || success == numResolver_) {
+  if (success > 0) {
     return 1;
   }
   else if (error == numResolver_) {
