From 6a1565c8f4cbeda0b11336c55439164dabe66b34 Mon Sep 17 00:00:00 2001
From: Gerald Schnabel <gschnabel@gmx.de>
Date: Mon, 13 Sep 2021 22:23:49 +0200
Subject: [PATCH] Fix -Wincompatible-pointer-types

---
 src/bimp-manipulations-gui.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/bimp-manipulations-gui.c b/src/bimp-manipulations-gui.c
index 336e51a..36ff7ba 100644
--- a/src/bimp-manipulations-gui.c
+++ b/src/bimp-manipulations-gui.c
@@ -45,39 +45,39 @@ void bimp_open_editwindow(manipulation man, gboolean first_time)
     
     if (man->type == MANIP_RESIZE) {
         gui = bimp_resize_gui_new((resize_settings)(man->settings));
-        save = &bimp_resize_save;
+        save = (void (*)(void *))&bimp_resize_save;
     }
     else if (man->type == MANIP_CROP) {
         gui = bimp_crop_gui_new((crop_settings)(man->settings));
-        save = &bimp_crop_save;
+        save = (void (*)(void *))&bimp_crop_save;
     }
     else if (man->type == MANIP_FLIPROTATE) {
         gui = bimp_fliprotate_gui_new((fliprotate_settings)(man->settings));
-        save = &bimp_fliprotate_save;
+        save = (void (*)(void *))&bimp_fliprotate_save;
     }
     else if (man->type == MANIP_COLOR) {
         gui = bimp_color_gui_new((color_settings)(man->settings));
-        save = &bimp_color_save;
+        save = (void (*)(void *))&bimp_color_save;
     }
     else if (man->type == MANIP_SHARPBLUR) {
         gui = bimp_sharpblur_gui_new((sharpblur_settings)(man->settings));
-        save = &bimp_sharpblur_save;
+        save = (void (*)(void *))&bimp_sharpblur_save;
     }
     else if (man->type == MANIP_CHANGEFORMAT) {
         gui = bimp_changeformat_gui_new((changeformat_settings)(man->settings), window);
-        save = &bimp_changeformat_save;
+        save = (void (*)(void *))&bimp_changeformat_save;
     }
     else if (man->type == MANIP_WATERMARK) {
         gui = bimp_watermark_gui_new((watermark_settings)(man->settings));
-        save = &bimp_watermark_save;
+        save = (void (*)(void *))&bimp_watermark_save;
     }
     else if (man->type == MANIP_RENAME) {
         gui = bimp_rename_gui_new((rename_settings)(man->settings), window);
-        save = &bimp_rename_save;
+        save = (void (*)(void *))&bimp_rename_save;
     }
     else if (man->type == MANIP_USERDEF) {
         gui = bimp_userdef_gui_new((userdef_settings)(man->settings), window);
-        save = &bimp_userdef_save;
+        save = (void (*)(void *))&bimp_userdef_save;
     }
     else {
         gtk_widget_destroy (window);
