From 279099c7c212f38cf7d4b504e58824fb6ebcdffe Mon Sep 17 00:00:00 2001
From: Gerald Schnabel <gschnabel@gmx.de>
Date: Mon, 13 Sep 2021 22:24:41 +0200
Subject: [PATCH] Fix -Wformat-overflow=

---
 src/bimp-operate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bimp-operate.c b/src/bimp-operate.c
index f030bda..a20f354 100644
--- a/src/bimp-operate.c
+++ b/src/bimp-operate.c
@@ -1007,7 +1007,7 @@ static gboolean apply_rename(rename_settings settings, image_output out, char* o
     // same thing for count and datetime 
     
     if(strstr(out->filename, RENAME_KEY_COUNT) != NULL)    {
-        char strcount[5];
+        char strcount[12];
         sprintf(strcount, "%i", processed_count + 1);
         out->filename = str_replace(out->filename, RENAME_KEY_COUNT, strcount);
     }
