From 6b3cd0c853fa95d7691f05a1018731e3306a160c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tuomo=20Syv=C3=A4nper=C3=A4?= <tuomo.syvanpera@gmail.com>
Date: Thu, 5 Mar 2020 12:02:35 +0200
Subject: [PATCH] when moving tabs and focus

---

--- tabbed.c
+++ tabbed.c
@@ -791,9 +791,9 @@ movetab(const Arg *arg)
 	if (sel < 0)
 		return;
 
-	c = (sel + arg->i) % nclients;
-	if (c < 0)
-		c += nclients;
+	c = sel + arg->i;
+	if (c < 0 || c == nclients)
+    return;
 
 	if (c == sel)
 		return;

@@ -902,10 +902,8 @@ rotate(const Arg *arg)
 	} else if (sel > -1) {
 		/* Rotating in an arg->i step around the clients. */
 		nsel = sel + arg->i;
-		while (nsel >= nclients)
-			nsel -= nclients;
-		while (nsel < 0)
-			nsel += nclients;
+    if (nsel < 0 || nsel >= nclients)
+      nsel = sel;
 		focus(nsel);
 	}
 }
