--- config.def.h	2018-09-27 10:05:24.000000000 +0200
+++ config.def.h	2018-09-27 10:09:32.011868287 +0200
@@ -1,13 +1,13 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const char font[]        = "monospace:size=9";
-static const char* normbgcolor  = "#222222";
-static const char* normfgcolor  = "#cccccc";
-static const char* selbgcolor   = "#555555";
-static const char* selfgcolor   = "#ffffff";
-static const char* urgbgcolor   = "#111111";
-static const char* urgfgcolor   = "#cc0000";
+static const char font[]        = "monospace:size=13";
+static const char* normbgcolor  = "#f2f2f2";
+static const char* normfgcolor  = "#2d2d2d";
+static const char* selbgcolor   = "#fef7e4";
+static const char* selfgcolor   = "#2d2d2d";
+static const char* urgbgcolor   = "#fafafa";
+static const char* urgfgcolor   = "#2d2d2d";
 static const char before[]      = "<";
 static const char after[]       = ">";
 static const char titletrim[]   = "...";

@@ -57,7 +57,7 @@ static Key keys[] = {
 	{ MODKEY,               XK_9,      move,        { .i = 8 } },
 	{ MODKEY,               XK_0,      move,        { .i = 9 } },
 
-	{ MODKEY,               XK_q,      killclient,  { 0 } },
+	{ MODKEY|ShiftMask,               XK_q,      killclient,  { 0 } },
 
 	{ MODKEY,               XK_u,      focusurgent, { 0 } },
 	{ MODKEY|ShiftMask,     XK_u,      toggle,      { .v = (void*) &urgentswitch } },
