--- main.c
+++ main.c
@@ -1325,7 +1325,7 @@ static int list_windows (Display *disp)
 	    /* geometry */
         XGetGeometry (disp, client_list[i], &junkroot, &junkx, &junky,
                           &wwidth, &wheight, &bw, &depth);
-        XTranslateCoordinates (disp, client_list[i], junkroot, junkx, junky,
+        XTranslateCoordinates (disp, client_list[i], junkroot, -junkx, -junky,
                                &x, &y, &junkroot);
       
         /* special desktop ID -1 means "all desktops", so we 
