--- configure	2018-11-11 11:56:43.000000000 +0100
+++ configure	2018-11-11 12:25:29.471856563 +0100
@@ -24089,8 +24089,9 @@
 fi
 fi
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for glib-networking (GIO TLS implementation)" >&5
-$as_echo_n "checking for glib-networking (GIO TLS implementation)... " >&6; }
+if test "x$with_webkit" != xno; then
+ { $as_echo "$as_me:${as_lineno-$LINENO}: checking for glib-networking (GIO TLS implementation)" >&5
+ $as_echo_n "checking for glib-networking (GIO TLS implementation)... " >&6; }
 gimp_save_CFLAGS="$CFLAGS"
 gimp_save_LIBS="$LIBS"
 CFLAGS="$CFLAGS $GIO_CFLAGS"
@@ -24141,6 +24142,7 @@
          HTTPS is becoming the expected standard and should not be considered
          optional anymore."
 fi
+fi
 
 # AM_PATH_GLIB_2_0 already fills in the variable GLIB_COMPILE_RESOURCES.
 # Unfortunately it looks it up using $PKG_CONFIG, which search in
 
