From 896f58c20aa2ed8ce884ecd7eeae6fbfefc4db29 Mon Sep 17 00:00:00 2001
From: Viktor Szakats <vszakats@users.noreply.github.com>
Date: Sun, 7 Jun 2020 19:28:17 +0200
Subject: [PATCH] udevil: recognize HFS+ as a filesystem

---
 src/udevil.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udevil.c b/src/udevil.c
index bab80e9..69a2d3f 100644
--- a/src/udevil.c
+++ b/src/udevil.c
@@ -632,7 +632,7 @@ void restore_privileges()
 char* get_known_filesystems()
 {
     // common types
-    char* list = g_strdup( "btrfs,ext2,ext3,ext4,udf,iso9660,xfs,jfs,nilfs,reiserfs,reiser4,msdos,umsdos,vfat,exfat,ntfs" );
+    char* list = g_strdup( "btrfs,ext2,ext3,ext4,udf,iso9660,xfs,jfs,nilfs,reiserfs,reiser4,msdos,umsdos,vfat,exfat,ntfs,hfsplus" );
 
     // get additional types from files
     static const char *type_files[] = { "/proc/filesystems", "/etc/filesystems", NULL };
