--- src/ptk/ptk-clipboard.c	2018-03-04 17:04:33.000000000 +0100
+++ src/ptk/ptk-clipboard.c	2021-03-05 17:37:56.851188199 +0100
@@ -184,14 +184,14 @@
     new_target = g_new0( GtkTargetEntry, 1 );
     new_target->target = "x-special/gnome-copied-files";
     #endif
-    g_memmove( &(targets[ n_targets - 2 ]), new_target, sizeof (GtkTargetEntry));
+    memmove( &(targets[ n_targets - 2 ]), new_target, sizeof (GtkTargetEntry));
     #if 0
     new_target = gtk_target_entry_new( "text/uri-list", 0, 0 );
     #else
     new_target = g_new0( GtkTargetEntry, 1 );
     new_target->target = "text/uri-list";
     #endif
-    g_memmove( &(targets[ n_targets - 1 ]), new_target, sizeof (GtkTargetEntry));
+    memmove( &(targets[ n_targets - 1 ]), new_target, sizeof (GtkTargetEntry));
 
     gtk_target_list_unref ( target_list );
 
@@ -237,14 +237,14 @@
     new_target = g_new0( GtkTargetEntry, 1 );
     new_target->target = "x-special/gnome-copied-files";
     #endif
-    g_memmove( &(targets[ n_targets - 2 ]), new_target, sizeof (GtkTargetEntry));
+    memmove( &(targets[ n_targets - 2 ]), new_target, sizeof (GtkTargetEntry));
     #if 0
     new_target = gtk_target_entry_new( "text/uri-list", 0, 0 );
     #else
     new_target = g_new0( GtkTargetEntry, 1 );
     new_target->target = "text/uri-list";
     #endif
-    g_memmove( &(targets[ n_targets - 1 ]), new_target, sizeof (GtkTargetEntry));
+    memmove( &(targets[ n_targets - 1 ]), new_target, sizeof (GtkTargetEntry));
 
     gtk_target_list_unref ( target_list );
 
