--- src/vfs/vfs-thumbnail-loader.c	2018-03-04 17:04:33.000000000 +0100
+++ src/vfs/vfs-thumbnail-loader.c	2020-11-11 09:21:58.102842816 +0100
@@ -418,7 +418,7 @@
     strcpy( ( file_name + 32 ), ".png" );
 
     thumbnail_file = g_build_filename( g_get_home_dir(),
-                                       ".thumbnails/normal",
+                                       ".cache/thumbnails/normal",
                                        file_name, NULL );
 
     if( G_UNLIKELY( 0 == mtime ) )
@@ -547,7 +547,7 @@
 void vfs_thumbnail_init()
 {
     char* dir;
-    dir = g_build_filename( g_get_home_dir(), ".thumbnails/normal", NULL );
+    dir = g_build_filename( g_get_home_dir(), ".cache/thumbnails/normal", NULL );
 
     if( G_LIKELY( g_file_test( dir, G_FILE_TEST_IS_DIR ) ) )
         chmod( dir, 0700 );
