--- CMakeLists.txt    2021-03-01 07:37:20.447294425 -0500
+++ CMakeLists.txt    2021-03-01 07:39:40.222030571 -0500
@@ -18,6 +18,9 @@
   message(FATAL_ERROR "Compiler with C++11 support is required")
 endif (HAS_CXX11)
 
+# Add support for more install configuration
+include(GNUInstallDirs)
+
 # Look for dependencies
 find_package(X11 REQUIRED)
 find_package(Iconv REQUIRED)
@@ -116,7 +119,7 @@
   -DFEATURES="${pekwm_FEATURES}"
   -DBINDIR="${CMAKE_INSTALL_PREFIX}/bin"
   -DDATADIR="${CMAKE_INSTALL_PREFIX}/share"
-  -DSYSCONFDIR="${CMAKE_INSTALL_PREFIX}/etc/pekwm")
+  -DSYSCONFDIR="${CMAKE_INSTALL_FULL_SYSCONFDIR}/pekwm")
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH True)
 
 # Subdirectories
--- data/CMakeLists.txt    2021-03-01 07:37:20.447294425 -0500
+++ data/CMakeLists.txt    2021-03-01 07:39:40.222030571 -0500
@@ -17,4 +17,4 @@
   panel
   start
   vars
-  DESTINATION etc/pekwm/)
+  DESTINATION ${CMAKE_INSTALL_FULL_SYSCONFDIR}/pekwm/)
